src/vcpkg/portfileprovider.cpp

```diff
@@ -108,22 +109,27 @@ namespace vcpkg
 
             virtual ExpectedL<Version> get_baseline_version(StringView port_name) const override
             {
-                auto it = m_baseline_cache.find(port_name);
-                if (it != m_baseline_cache.end())
-                {
-                    return it->second;
-                }
-                else
-                {
-                    auto version = registry_set.baseline_for_port(port_name);
-                    m_baseline_cache.emplace(port_name.to_string(), version);
-                    return version;
-                }
+                return m_baseline_cache.get_lazy(port_name, [this, port_name]() -> ExpectedL<Version> {
+                    auto maybe_maybe_version = registry_set.baseline_for_port(port_name);
+                    auto maybe_version = maybe_maybe_version.get();
+                    if (!maybe_version)
+                    {
+                        return std::move(maybe_maybe_version).error();
+                    }
+
+                    auto version = maybe_version->get();
+                    if (!version)
+                    {
+                        return msg::format_error(msgPortNotInBaseline, msg::package_name = port_name);
```

Comment:
I looked at doing this but I'm not sure what we would print; registries don't have names and we don't currently have a way in console output to refer to a particular one. I think it would be good to emit the error attributed to the JSON document declaring the registry someday but I think that's outside the scope of this PR.